### PR TITLE
Create initial enigma spec and class

### PR DESCRIPTION
**Pull request by sjmann2**
This creates the spec file and enigma class file. It includes passing encrypt and decrypt methods. There's room to refactor encrypt and decrypt and pull code out into helper methods. It passes the edge case test of uppercase letters, but still needs code to pass the edge case test of special characters.

### CODE REVIEW

File: lib/enigma.rb [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+  def offset_generator(date)
+    squared_date = (date.to_i ** 2).to_s
+    squared_date[-4..-1]
+  end
+
+  def encrypt(message, key = key_generator, date = date_generator)
+    offset = offset_generator(date)
+    message.downcase!
+    if message.include?(!@characters)
```
Comment (sjmann2) [inline lib/enigma.rb]: You meant to delete this half written code, can you delete it and and commit?

**Comment by sjmann2** [comment]
Great first draft, get in there and refactor those long methods doing more than one thing!
